web/api/v1/output.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/prometheus/prometheus/promql"
+)
+
+// modifier is a function that alters the value of a query result.
+type modifier struct {
+	name  string
+	apply func([]string, *promql.Result) (*promql.Result, error)
+}
+
+// modifiers is an ordered list of registered modifiers.
```

Comment:
This seems overengineered, there's only going to be 2 of them.

Reply:
sort, sort_desc, sort_label, sort_label_desc, limit

Reply:
And we can later maybe extend to have the  ASAP smoothing and others

Reply:
That's only 2: sort and limit.

Think of it this way, if I list both sort and sort_desc which wins? Those can't be the url parameter names, those need to be all handled by values of one parameter.

Reply:
What about:

&sort=1 -> sort asc
&sort=-1 -> sort desc

for labels:

&sort=-job,sort=instance
-job = job, desc

then there is the sort by label-value-as-a-bucket

Reply:
Regarding the order:

I think we need to: 1. sort by value 2. sort by label 3. limit

That way the things will be sorted out by value in a label

Instead of by label in a value which does not make sense unless we expect to have a lot of == values, which is not the case.

Reply:
> I think we need to: 1. sort by value 2. sort by label 

Why would we add that sort of restriction? This is not the place to design it though, the point is there's no need to make this so generic.